Lib/fontTools/misc/xmlWriter.py

```diff
@@ -4,8 +4,19 @@
 import sys
 import os
 import string
+import logging
 
 INDENT = "  "
+TTX_LOG = logging.getLogger("fontTools.ttx")
+REPLACEMENT = "?"
+ILLEGAL_CHARS = dict.fromkeys(
+    [chr(c) for c in range(0x00, 0x09)]
+    + [chr(0x0B), chr(0x0C)]
+    + [chr(c) for c in range(0x0E, 0x20)]
+    + [chr(0xFFFE), chr(0xFFFF)],
+    REPLACEMENT,
+)
+ILLEGAL_TRANS = str.maketrans(ILLEGAL_CHARS)
```

Comment:
we don't actuallt need to call str.maketrans, that just returns a plain dict with integers (unicode codepoint) as keys and the replacement string(s) as values, so you could simply do

```suggestion
ILLEGAL_XML_CHARS = dict.fromkeys(
    itertools.chain(
        range(0x00, 0x09),
        (0x0B, 0x0C),
        range(0x0E, 0x20),
        (0xFFFE, 0xFFFF),
    ),
    REPLACEMENT,
)
```

and then you'd use it below like

```
newData = data.translate(ILLEGAL_XML_CHARS)
```

Reply:
Could you maybe also add a link to the spec? 

https://www.w3.org/TR/xml/#charsets

Reply:
> we don't actually need to call str.maketrans

Doh! :)

> Could you maybe also add a link to the spec?
> 
> https://www.w3.org/TR/xml/#charsets

Added in new `escape` docstring.